flink-connectors/flink-connector-aws-kinesis-data-streams/src/main/java/org/apache/flink/connector/kinesis/sink/KinesisDataStreamsSink.java

```diff
@@ -111,7 +111,7 @@
         return new KinesisDataStreamsSinkBuilder<>();
     }
 
-    @Experimental
+    @Internal
```

Comment:
Having an `@Internal` public method on a `@Public` class seems like a bad design to me. How does this annotation stop users using it? Is this smell inherited from the base `SInk` or something we have added here

Reply:
The user is never expected to call createWriter right, I guess they're effectively prevented from doing so because they'll never have a InitContext object. I thought the createWriter was only here for the sink operator to use...

Reply:
i.e.
```
KinesisDataStreamsSink.createWriter(org.apache.flink.api.connector.sink.Sink$InitContext, java.util.List): Argument leaf type org.apache.flink.api.connector.sink.Sink$InitContext does not satisfy: reside outside of package 'org.apache.flink..' or annotated with @Public or annotated with @PublicEvolving or annotated with @Deprecated
KinesisDataStreamsSink.createWriter(org.apache.flink.api.connector.sink.Sink$InitContext, java.util.List): Returned leaf type org.apache.flink.api.connector.sink.SinkWriter does not satisfy: reside outside of package 'org.apache.flink..' or annotated with @Public or annotated with @PublicEvolving or annotated with @Deprecated
```

Reply:
Yes but it seems like a design smell. Why have a public method on a contract that is not expected to be used? Is this the same with other Sinks?

Reply:
I see this is an artifact of the `Sink` design as `KafkaSink` is the same